### PR TITLE
Toposort Java Refactor

### PR DESCRIPTION
**Pull request by HarryWangATX**

### CODE REVIEW

File: content/4_Gold/TopoSort.mdx [hunk 1]
```diff
@@ -412,6 +414,13 @@ static void topological_sort() {
 
 </LanguageSection>
 
+<Info title="Pro Tip">
+
+If the array containing the end order does not equal to the number of elements that needed to be sorted, then 
+there is a cycle in the graph constructed by `edges`.
```
Comment (dongliu0426) [inline content/4_Gold/TopoSort.mdx:420]:
```suggestion
If the array containing the end order does not equal to the number of elements that needed to be sorted, then 
there is a cycle in the graph.
```

**Review by dongliu0426 — APPROVED** [review]

File: content/4_Gold/TopoSort.mdx [hunk 2]
```diff
@@ -164,7 +164,7 @@ public class CourseSchedule {
 		Collections.reverse(topo);
 	}
 	public static void dfs(int v) { //not always lexigraphically least
```
Comment (maggie-j-liu) [inline content/4_Gold/TopoSort.mdx:166]:
```suggestion
	public static void dfs(int v) { //not always lexicographically least
```

File: content/4_Gold/TopoSort.mdx [hunk 3]
```diff
@@ -121,30 +121,30 @@ import java.util.*;
 import java.io.*;
 
 public class CourseSchedule {
-	public static ArrayList <Integer> g[];
-	public static ArrayList <Integer> topo = new ArrayList < Integer > ();
+	public static List<Integer>[] g;
```
Comment (SansPapyrus683) [inline content/4_Gold/TopoSort.mdx:124]: if you're gonna refactors at least rename this to something more explicit like `neighbors`

File: content/4_Gold/TopoSort.mdx [hunk 4]
```diff
@@ -382,13 +383,13 @@ void compute() {
 <JavaSection>
 
 ```java
-static int in_degree[];
-static ArrayList<Integer> edge[]; //adjacency list
+static int[] in_degree;
```
Comment (SansPapyrus683) [inline content/4_Gold/TopoSort.mdx:386]: camelCase? (there's some other places that don't have it, be sure to do that)
also uh wrap this in a class

File: content/4_Gold/TopoSort.mdx [hunk 5]
```diff
@@ -573,9 +582,9 @@ import java.io.*;
 //longest_path
 public class Main {
 
-	static int prev_flight[], dist[], in_degree[];
-	static ArrayList<Integer> edge[];
-	static ArrayList<Integer> backEdge[];
+	static int[] prev_flight, dist, in_degree;
```
Comment (SansPapyrus683) [inline content/4_Gold/TopoSort.mdx:585]: camelCase

File: content/4_Gold/TopoSort.mdx [hunk 6]
```diff
@@ -644,7 +654,7 @@ public class Main {
 
 		PrintWriter pw = new PrintWriter(System.out);
 
-		ArrayDeque<Integer> answer = new ArrayDeque<Integer>(); //Acts as a Stack
+		ArrayDeque<Integer> answer = new ArrayDeque<>(); //Acts as a Stack
```
Comment (SansPapyrus683) [inline content/4_Gold/TopoSort.mdx:657]:
```suggestion
		ArrayDeque<Integer> answer = new ArrayDeque<>();  // Acts as a Stack
```

File: content/4_Gold/TopoSort.mdx [hunk 7]
```diff
@@ -121,30 +121,30 @@ import java.util.*;
...
 			g[a].add(b);
 		}
+
 		compute();
-		int ind[] = new int[N];
+		int[] ind = new int[N];
+		
 		for (int i = 0; i < N; i++) ind[topo.get(i)] = i;
 		for (int i = 0; i < N; i++)
 			for (int j: g[i])
```
Comment (SansPapyrus683) [inline content/4_Gold/TopoSort.mdx:150]:
```suggestion
			for (int j : g[i])
```

**Review by SansPapyrus683 — APPROVED** [review]
better than the original but still some stuff that can be improved

File: content/4_Gold/TopoSort.mdx [hunk 8]
```diff
@@ -412,6 +414,13 @@ static void topological_sort() {
 
 </LanguageSection>
 
+<Info title="Pro Tip">
+
+If the array containing the end order does not equal to the number of elements that needed to be sorted, then
```
Comment (maggie-j-liu) [inline content/4_Gold/TopoSort.mdx:419]:
```suggestion
If the length of the array containing the end order does not equal the number of elements that need to be sorted, then 
```
is this what you meant?

File: content/4_Gold/TopoSort.mdx [hunk 9]
```diff
@@ -644,7 +652,7 @@ public class Main {
 
 		PrintWriter pw = new PrintWriter(System.out);
 
-		ArrayDeque<Integer> answer = new ArrayDeque<Integer>(); //Acts as a Stack
+		ArrayDeque<Integer> answer = new ArrayDeque<>(); //Acts as a Stack
 		int temp = N-1;
```
Comment (maggie-j-liu) [inline content/4_Gold/TopoSort.mdx:656]:
```suggestion
		int temp = N - 1;
```
can you add spaces in all places that need them (can't add suggestions but lines 559, 561, 668, 670 and there's probably more that i didn't find)

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]

**Review by maggie-j-liu — APPROVED** [review]

File: content/4_Gold/TopoSort.mdx (hunk 3 above)
Comment (SansPapyrus683) [reply content/4_Gold/TopoSort.mdx]: uh, not resolved?

**Pull request merged**